src/common/reconciler.go

```diff
@@ -58,234 +77,201 @@ func (r *ServiceReconciler) Reconcile(registration opslevel_jq_parser.ServiceReg
 }
 
-// This function has 4 outcomes that can happen while looping over the aliases list
+// lookupService has 4 outcomes that can happen while looping over the aliases list
 // serviceAliasesResult_NoAliasesMatched - means that all API calls succeeded and none of the aliases matched an existing service
 // serviceAliasesResult_AliasMatched - means that all the API calls succeeded and a single service was found matching 1 of N aliases
 // serviceAliasesResult_MultipleServicesFound - means that all API calls succeeded but multiple services were returning means the list of aliases does not definitively describe a single service and might be a configuration problem
 // serviceAliasesResult_APIErrorHappened - means that 1 of N aliases got a 4xx/5xx and thereforce we cannot say 100% that the services doesn't exist
-// serviceAliasesResult_FoundServiceNoAlias - means that a service was found but that service has no alias (this should not be possible and can only happen from a bad code change.)
 func (r *ServiceReconciler) lookupService(registration opslevel_jq_parser.ServiceRegistration) (*opslevel.Service, serviceAliasesResult) {
-	var gotError error
-	foundServices := map[string]*opslevel.Service{}
+	var foundService *opslevel.Service
 	for _, alias := range registration.Aliases {
-		foundService, err := r.client.GetService(alias)
+		gotService, err := r.client.GetService(alias)
 		if err != nil {
-			gotError = err
 			log.Warn().Err(err).Msgf("got an error when trying to get service with alias '%s'", alias)
-		} else if foundService == nil {
-			log.Warn().Msgf("unexpected happened: got service with alias '%s' but the result is nil", alias)
-		} else if foundService.Id == "" {
-			log.Warn().Msgf("unexpected happened: got service with alias '%s' but the result has no ID", alias)
-		} else {
-			// happy path
-			foundServices[string(foundService.Id)] = foundService
+			return nil, serviceAliasesResult_APIErrorHappened
+		} else if gotService == nil {
+			log.Debug().Msgf("did not find a service with alias '%s'", alias)
+			continue
+		} else if foundService != nil {
+			log.Warn().Msgf("found another service with the same alias '%s' (%s)", alias, gotService.Id)
+			return nil, serviceAliasesResult_MultipleServicesFound
 		}
+		// happy path
+		foundService = gotService
 	}
-	if gotError != nil {
-		return nil, serviceAliasesResult_APIErrorHappened
-	}
-	foundServicesCount := len(foundServices)
-	if foundServicesCount == 1 {
-		keys := maps.Keys(foundServices)
-		if len(keys) == 0 {
-			return nil, serviceAliasesResult_FoundServiceNoAlias
-		}
-		key := keys[0]
-		return foundServices[key], serviceAliasesResult_AliasMatched
-	} else if foundServicesCount > 1 {
-		return nil, serviceAliasesResult_MultipleServicesFound
-	} else {
+	if foundService == nil {
 		return nil, serviceAliasesResult_NoAliasesMatched
 	}
-}
-
-func (r *ServiceReconciler) handleService(registration opslevel_jq_parser.ServiceRegistration) (*opslevel.Service, error) {
```

Comment:
I have to assume this serves a function. Let's keep this and work toward removing it if possible / necessary rather than yanking it out wholesale